src/swap/getFeeCurrency.tsx

```diff
@@ -1,6 +1,20 @@
-import { TransactionRequestCIP42 } from 'node_modules/viem/_types/chains/celo/types'
+import { TransactionRequest } from 'src/viem/prepareTransactions'
+import { Address } from 'viem'
 
-export function getFeeCurrency(preparedTransactions: TransactionRequestCIP42[]) {
-  // The prepared transactions always use the same fee currency
-  return preparedTransactions[0].feeCurrency
+export function getFeeCurrency(preparedTransactions: TransactionRequest[]): Address | undefined
+export function getFeeCurrency(preparedTransaction: TransactionRequest): Address | undefined
+export function getFeeCurrency(x: TransactionRequest[] | TransactionRequest): Address | undefined {
+  let preparedTransaction: TransactionRequest
+  if (Array.isArray(x)) {
+    // The prepared transactions always use the same fee currency
+    preparedTransaction = x[0]
```

Comment:
nit: might be worth throwing an error if they don't

Reply:
Yep was thinking about this too. Adding it now. Thanks.